sos/collector/transports/control_persist.py

```diff
@@ -193,6 +193,12 @@ def remote_exec(self):
                             f"{self.opts.ssh_user}@{self.address}")
         return self.ssh_cmd
 
+    def _copy_file_to_remote(self, fname, dest):
+        cmd = (f"/usr/bin/scp -oControlPath={self.control_path} "
+               f"{fname} {self.opts.ssh_user}@{self.address}:{dest}")
+        res = sos_get_command_output(cmd)
```

Comment:
sos_get_command_output() uses TIMEOUT_DEFAULT, which is 5 minutes. Since you run this for 5 attempts, we may be waiting for a long time. Would it be worth to reduce the timeout in this call, or the number of attempts in copy_file_to_remote()?

Reply:
I'm thinking 10 seconds on control persist, and 15 seconds on juju, oc and saltstack, just to handle any extra latency that might be introduced or delays in retrieving jobs from message queues. I'm not dead set on the values if you think more or less time would be better.

Thoughts?

Reply:
These numbers sound good to me, I think it will reduce the wait a lot in extreme cases

Reply:
I went with the above listed timeouts, and 3 total attempts/retries. So control persist should only have 30 seconds max before failing to copy a config, and oc, juju and saltstack 45 seconds max.